### PR TITLE
Move src/go contents to base director

### LINKED ISSUE #24: get rid of /src/go

### PR DESCRIPTION
**Pull request by asalkeld**
closes #24

### CODE REVIEW

**Comment by mjudeikis** [comment]
strange error. Just to check if not a flake:
/test build

**Comment by mjudeikis** [comment]
```
error: error copying source path /go/src/github.com/openshift/azure-misc///.
```
something fishy here

**Comment by kargakis** [comment]
We merged the release changes, unclear what's the issue here.

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /.vscode
 /env
-/src/go/azure-purge
-/src/go/prometheus-sidecar
+/azure-purge
+/prometheus-sidecar
```
Comment (mjudeikis) [inline .gitignore]: I think this is an error. It ignores folder, and not binary.

**Comment by mjudeikis** [comment]
`ERROR: Error creating dir "/tmp/build/inputs/azure-purge": mkdir /tmp/build/inputs/azure-purge: not a directory`

This gives an error that it's not a dir, so it is a file. And this is same file name as binary. Can it be binary build created binary in a wrong location? Or was passed down the pipeline

File: .gitignore (hunk 1 above)
Comment (asalkeld) [reply .gitignore]: thanks, i'll fix that

**Comment by asalkeld** [comment]
Please note the outstanding PR for release. I missed a reference to src/go that needs to be merged.

**Comment by asalkeld** [comment]
I have reposted the reverted release PR here: https://github.com/openshift/release/pull/1632

**Comment by kargakis** [comment]
```
  source:
    images:
    - as: null
      from:
        kind: ImageStreamTag
        name: pipeline:bin
      paths:
      - destinationDir: .
        sourcePath: /go/src/github.com/openshift/azure-misc/prometheus-sidecar/prometheus-sidecar
    - as:
      - prometheus-sidecar
      from:
        kind: ImageStreamTag
        name: pipeline:root
      paths: null
    - as: null
      from:
        kind: ImageStreamTag
        name: pipeline:src
      paths:
      - destinationDir: .
        sourcePath: /go/src/github.com/openshift/azure-misc///.
    type: Image
```
This is the build created during the test. I think the problem is that the directories that are get copied by `pipeline:src` (in order to get the Dockerfile) collide with the binary that is getting copied by `pipeline:bin`.

File: azure-purge/Makefile [hunk 2]
```diff
@@ -2,6 +2,7 @@ PURGE_IMAGE ?= quay.io/openshift-on-azure/azure-purge:latest
 
 azure-purge:
 	go build ./cmd/azure-purge
```
Comment (kargakis) [inline azure-purge/Makefile]: go build -o azure-purge-bin ./cmd/azure-purge

File: prometheus-sidecar/Makefile [hunk 3]
```diff
@@ -2,6 +2,7 @@ PROMETHEUS_SIDECAR_IMAGE ?= quay.io/openshift-on-azure/prometheus-sidecar:latest
 
 prometheus-sidecar: generate
 	go build ./cmd/prometheus-sidecar
```
Comment (kargakis) [inline prometheus-sidecar/Makefile]: go build -o prometheus-sidecar-bin ./cmd/prometheus-sidecar

Comment (asalkeld) [reply prometheus-sidecar/Makefile]: done

**Pull request merged**